mitmproxy/addons/session.py

```diff
@@ -0,0 +1,75 @@
+import tempfile
+import sqlite3
+import os
+
+from mitmproxy.exceptions import SessionLoadException
+from mitmproxy.utils.data import pkg_data
+
+
+# Could be implemented using async libraries
+class SessionDB:
+    """
+    This class wraps connection to DB
+    for Sessions and handles creation,
+    retrieving and insertion in tables.
+    """
+
+    def __init__(self, db_path=None):
+        """
+        Connect to an already existing database,
+        or create a new one with optional path.
+        :param db_path:
+        """
+        self.temp = None
+        self.con = None
+        if db_path is not None and os.path.isfile(db_path):
+            self._load_session(db_path)
+        else:
+            if db_path:
+                path = db_path
+            else:
+                # We use tempfile only to generate a path, since we demand file creation to sqlite, and removal to os.
+                self.temp = tempfile.NamedTemporaryFile()
```

Comment:
This creates a file on the file system and returns you an opened file. It's probably not what we want when we're creating a database _de nuovo_. I suggest that you create a temporary directory instead, and then join the tmpdir path with some filename to get the database file path. 

This might be the root cause for the Windows unit test failure.